hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationMonitor.java

```diff
@@ -212,6 +214,50 @@ long getLastMemberHeartbeatMillis(Address memberAddress) {
         return heartbeat == null ? 0 : heartbeat.get();
     }
 
+    private abstract class MonitorTask implements Runnable {
+        @Override
+        public void run() {
+            try {
+                run0();
+            } catch (Throwable t) {
+                inspectOutputMemoryError(t);
+                logger.severe(t);
+            }
+        }
+
+        protected abstract void run0();
+    }
+
+    abstract class FixedRateMonitorTask implements Runnable {
+        final long periodMillis;
+        private long expectedNextMillis = System.currentTimeMillis();
+
+        FixedRateMonitorTask(long periodMillis) {
+            this.periodMillis = periodMillis;
+        }
+
+        @Override
+        public void run() {
+            long currentTimeMillis = System.currentTimeMillis();
+
+            try {
+                if (expectedNextMillis + MAX_DELAY_MILLIS < currentTimeMillis) {
+                    logger.warning(getClass().getSimpleName() + " delayed " + (currentTimeMillis - expectedNextMillis) + " ms");
```

Comment:
Would it contribute anything to triage the delay causes into "last task execution took too long" and "next task failed to be scheduled on time"?